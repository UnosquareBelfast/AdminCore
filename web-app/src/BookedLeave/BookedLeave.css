.LeaveTrackerList {
    width:30%;
    margin-top: 20px;
    align-items: flex-start;
}