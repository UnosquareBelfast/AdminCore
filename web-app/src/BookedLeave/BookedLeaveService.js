export default class BookedLeaveService {
    
    constructor(domain) {
        this.domain = domain || 'http://localhost';
        this.getHolidays = this.getHolidays.bind(this);
        this.updateHoliday = this.updateHoliday.bind(this);
        this.updateHolidays = this.updateHolidays.bind(this);
    }

    // TODO: We need to wrap our requests to check if the token is expired
    getHolidays(employeeId){
        return fetch(`${this.domain}/holidays/findByEmployeeId/` + employeeId, { 
            method: 'GET',
            headers: { "content-type": "Application/json"}
        })
        .then(this._checkStatus)
        .then(response => response.json())
        .then(res => {            
            return Promise.resolve(res);
        });
    }

    updateHoliday(holiday){
        return fetch(`${this.domain}/holidays/`, { 
            method: 'PUT',
            headers: { "content-type": "Application/json"},
            body: JSON.stringify({ holiday }),
        })
        .then(this._checkStatus)
        .then(response => response.json())
        .then(res => {            
            return Promise.resolve(res);
        });
    }

    updateHolidays(holidays){
        return fetch(`${this.domain}/holidays/updateMultiple`, { 
            method: 'PUT',
            headers: { "content-type": "Application/json"},
            body: JSON.stringify({ holidays }),
        })
        .then(this._checkStatus)
        .then(response => response.json())
        .then(res => {            
            return Promise.resolve(res);
        });
    }

    _checkStatus(response) {
        if (response.status >= 200 && response.status < 300) {
            return response
        } else {
            var error = new Error(response.statusText)
            error.response = response
            throw error
        }
    }
}