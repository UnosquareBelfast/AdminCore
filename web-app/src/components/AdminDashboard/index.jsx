import React from 'react';
import { Link } from 'react-router-dom';
import { InnerLayout } from './styled';

const AdminDashboard = () => {
  return (
    <InnerLayout>
      <h2>Admin Dashboard</h2>
      <div className="columnWrap">
        <div>
          <h3>Employees</h3>
          <Link to="/admin/employees">View Employees</Link>
        </div>
        <div>
          <h3>Holidays</h3>
          <Link to="/admin/holidays">View All Holidays</Link>
          <Link to="/admin/pendingHolidays">View Pending Holidays</Link>
        </div>
        <div>
          <h3>Clients</h3>
          <Link to="">View All Holidays</Link>
          <Link to="">View Pending Holidays</Link>
        </div>
        <div>
          <h3>Contracts</h3>
          <Link to="">View All Holidays</Link>
          <Link to="">View Pending Holidays</Link>
        </div>
      </div>
    </InnerLayout>
  );
};

export default AdminDashboard;
