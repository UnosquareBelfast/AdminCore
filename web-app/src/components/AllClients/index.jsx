import React, { Fragment } from 'react';
import { PropTypes as PT } from 'prop-types';
import container from './container';
import { Table, Button } from '../common';
import { ActiveDot, CTA } from './styled';
import employeeStatus from '../../utilities/employeeStatus';

export const AllClients = ({
  clients,
  viewClient,
  archive,
  createNewClient,
}) => {
  let clientList;
  if (clients.length > 0) {
    clientList = clients.map(client => (
      <tr key={client.clientId}>
        <td>{client.clientName}</td>
        <td>{client.clientStatusDescription}</td>
        <td>
          <ActiveDot active={client.clientStatusId === employeeStatus.ACTIVE} />
          {client.clientStatusId === employeeStatus.ACTIVE
            ? 'Active'
            : 'Inactive'}
        </td>
        <td>
          <button onClick={() => viewClient(client.clientId)}>
            Edit Client
          </button>
          {client.clientStatusId === employeeStatus.ACTIVE ? (
            <button className="success" onClick={() => archive(client)}>
              Archive
            </button>
          ) : (
            <button className="error" onClick={() => archive(client, true)}>
              Activate
            </button>
          )}
        </td>
      </tr>
    ));
  } else {
    clientList = (
      <tr>
        <td>No Clients</td>
        <td />
        <td />
        <td />
      </tr>
    );
  }
  return (
    <Fragment>
      <h2>All Clients</h2>
      <Table
        tableHeaders={['Name', 'Description', 'Status', ' ']}
        tableRows={clientList}
      />
      <CTA>
        <Button onClick={() => createNewClient()} label="New Client" />
      </CTA>
    </Fragment>
  );
};

AllClients.propTypes = {
  createNewClient: PT.func,
  clients: PT.array,
  archive: PT.func.isRequired,
  viewClient: PT.func.isRequired,
};

export default container(AllClients);
