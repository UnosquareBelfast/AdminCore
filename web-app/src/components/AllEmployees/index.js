import React, { Fragment } from 'react';
import { PropTypes as PT } from 'prop-types';
import { UserList, UserModal } from '../';
import { Button } from '../common';
import { CornerButton } from '../common_styled';
import container from './container';

const AllEmployees = ({ history, users, selectUser, selectedUser }) => {
  return (
    <Fragment>
      <CornerButton>
        <Button
          onClick={() => history.replace('/admin/employees/new')}
          label="New Employee"
        />
      </CornerButton>
      <UserModal
        user={selectedUser}
        closeModal={() => selectUser({})}
        history={history}
      />
      <h2>Employees</h2>
      <UserList
        history={history}
        users={users}
        columns={['fullName', 'email', 'role', 'location']}
        onRowClick={employee => selectUser(employee)}
      />
    </Fragment>
  );
};

AllEmployees.propTypes = {
  history: PT.object.isRequired,
  users: PT.array.isRequired,
  selectUser: PT.func.isRequired,
  selectedUser: PT.object.isRequired,
};

export default container(AllEmployees);
