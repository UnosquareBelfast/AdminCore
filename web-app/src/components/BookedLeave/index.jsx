import React from 'react';
import { PropTypes as PT } from 'prop-types';
import styles from './style.css';
import container from './container';

export const BookedLeave = (props) => {
  const formatDate = (date) => {
    const definedDate = new Date(date);
    const year = definedDate.getFullYear();
    const month = definedDate.toLocaleString('en-us', { month: 'long'});
    const day = definedDate.getDate();

    return year + '-' + month + '-' + day;
  };

  return (
    <div>
      {
        props.bookedHolidays.map(holiday => {
          return <span className={styles.LeaveTrackerList}>{formatDate(holiday.date)}</span>;
        })
      }
    </div>
  );
};

BookedLeave.propTypes = {
  bookedHolidays: PT.array,
};

export default container(BookedLeave);
