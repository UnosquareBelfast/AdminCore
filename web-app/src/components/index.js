export { default as AdminDashboard } from './AdminDashboard';
export { default as AllClients } from './AllClients';
export { default as AllHolidays } from './AllHolidays';
export { default as BigCalendarToolbar } from './BigCalendarToolbar';
export { default as BookedLeave } from './BookedLeave';
export { default as BookingCalendar } from './BookingCalendar';
export { default as BookingModal } from './BookingModal';
export { default as CreateClient } from './CreateClient';
export { default as CreateContract } from './CreateContract';
export { default as ClientModal } from './ClientModal';
export { default as CreateTeam } from './CreateTeam';
export { default as CreateUser } from './CreateUser';
export { default as Legend } from './Legend';
export { default as PendingHolidays } from './PendingHolidays';
export { default as RangeDayPicker } from './common/RangeDayPicker';
export { default as RequestHoliday } from './RequestHoliday';
export { default as TakenLeave } from './TakenLeave';
export { default as TeamView } from './TeamView';
export { default as UserDetails } from './UserDetails';
export { default as UserModal } from './UserModal';
export { default as ViewContracts } from './ViewContracts';
export { default as ViewTeams } from './ViewTeams';
export { default as HolidayModal } from './HolidayModal';
export { default as AllEmployees } from './AllEmployees';
export { default as DataTable } from './DataTable';
