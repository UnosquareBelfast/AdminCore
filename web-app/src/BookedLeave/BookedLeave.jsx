import React from 'react';
import Swal from 'sweetalert2';
import styles from './BookedLeave.css';
import Moment from 'moment';
import BookedLeaveService from './BookedLeaveService';

class BookedLeave extends React.Component {

    constructor(props){
        super(props);

        this.state = {
            bookedHolidays: []
        };

        this.BookedLeaveService = new BookedLeaveService();
    }

    componentDidMount(){
        this.BookedLeaveService.getHolidays(this.props.user.id)
        .then(response =>{
            const bookedHolidays = response.filter(hol => { return this.isDateInTheFuture(hol.date)});
            this.setState({bookedLeave: bookedHolidays});
        })
        .catch(error =>{
            Swal('Could not get booked holidays', error.message, 'error');
        })
    }

    formatDate(date){
        const definedDate = new Date(date);
        const year = definedDate.getFullYear();
        const month = definedDate.toLocaleString("en-us", { month: "long"});
        const day = definedDate.getDate();
            
        return year + '-' + month + '-' + day;
    }

    isDateInTheFuture(date){
        return Moment(date).isAfter(new Date());
    }

    render() {
      return (
        <div>
            {
                this.state.bookedHolidays.map(holiday => {                    
                    return <span className={styles.LeaveTrackerList}>{this.formatDate(holiday.date)}</span>                
                })
            }
        </div>
      );
    }
  }
  
  export default BookedLeave;