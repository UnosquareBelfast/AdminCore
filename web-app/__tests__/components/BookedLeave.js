import React from 'react';
import { shallow } from 'enzyme';
import { BookedLeave } from '../../src/components/BookedLeave';

describe('Booked Leave', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<BookedLeave bookedHolidays={[]} user={{id: 0}}/>); 
    expect(wrapper).toMatchSnapshot();
  });
});
