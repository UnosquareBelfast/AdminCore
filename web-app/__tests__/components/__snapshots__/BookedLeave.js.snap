// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booked Leave renders correctly 1`] = `<div />`;
